apis/v1alpha2/gateway_types.go

```diff
@@ -518,12 +518,19 @@ const (
 	// Support: Extended
 	IPAddressType AddressType = "IPAddress"
 
-	// An opaque identifier that represents a specific IP address. The
-	// interpretation of the name is dependent on the controller. For
-	// example, a "NamedAddress" might be a cloud-dependent identifier
-	// for a static or elastic IP.
+	// A hostname represents a DNS based ingress point. This is similar to the
+	// corresponding hostname field in Kubernetes load balancer status. For
+	// example, this concept may be used for cloud load balancers where a DNS
+	// name is used to expose a load balancer.
 	//
-	// Support: Implementation-specific
+	// Support: Extended
+	HostnameAddressType AddressType = "Hostname"
+
+	// A NamedAddress provides a way to reference a specific IP address by name.
+	// For example, this may be a name or other unique identifier that refers
+	// to a static IP address on a cloud provider.
+	//
+	// Support: Implementation-Specific
```

Comment:
Only leaving a note (no changes expected): Support in these cases will be based on a combination of the controller AND the cluster provider (IaaS, whatever) supporting and agreeing on some out-of-band API. This is abstract enough that we don't need to be prescriptive about it but we also have to be careful of users not using this in place of the other two more concrete types (for any reason).